### PR TITLE
Add LpPool-18 - add `ceil_mode` and `dilations` attributes

### PR DESCRIPTION
**Pull request by p-wysocki**
Signed-off-by: p-wysocki <przemyslaw.wysocki@intel.com>

### Description
Add `LpPool-18`, which now has `ceil_mode` and `dilations` attributes. Deprecate `auto_pad`, like it is done in the case of [AveragePool](https://github.com/onnx/onnx/blob/main/docs/Operators.md#averagepool) and [MaxPool](https://github.com/onnx/onnx/blob/main/docs/Operators.md#maxpool).

### Motivation and Context
- Partially solves https://github.com/onnx/onnx/issues/2290, but more discussion may be needed.

### Notes
- [LpPoolOpSchemaGenerator](https://github.com/onnx/onnx/blob/main/onnx/defs/nn/defs.cc#L541) uses [convPoolShapeInference](https://github.com/onnx/onnx/blob/main/onnx/defs/nn/defs.cc#L36), which [already accounts](https://github.com/onnx/onnx/blob/main/onnx/defs/nn/defs.cc#L176-L180) for `ceil_mode` attribute. As far as I understand, it makes adding the attribute to `ceil_mode` trivial, as it is done in this PR. The same goes for the `dilations` attribute.
- The tests for `LpPool` are non-existent (there is not such a file as `onnx/onnx/backend/test/case/node/lppool.py`). Is it intentional? Pooling operators seem to have a lot of common code, so maybe `LpPool` is considered to be covered by tests for `MaxPool` and `AveragePool`. I am not sure if it is intended or an overlook.
- The issue touches on the subject of alignment of Pooling operators. Their attributes are as follows:
image
A related PR concerns removing `storage_order` attribute from the MaxPool operator: https://github.com/onnx/onnx/pull/4533.

### CODE REVIEW

**Comment by kraiskil** [comment]
I don't see why explicit `LpPool` tests would be left out intentionally. After all, the differnt pools calculate different things.

In my experience, when writing a backend, such coverage tests are very valuable, partly as a safety net but also as additional documentation / clarification. I can not see any big cost in adding such tests, except of course for the initial effort of writing them.

File: onnx/version_converter/adapters/lppool_17_18.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include "onnx/version_converter/adapters/transformers.h"
+
+namespace ONNX_NAMESPACE {
+namespace version_conversion {
+
+class LpPool_17_18 final : public Adapter {
+ public:
+  explicit LpPool_17_18() : Adapter("LpPool", OpSetID(17), OpSetID(18)) {}
+
+  void adapt_lppool_17_18(std::shared_ptr<Graph>, Node* node) const {
```
Comment (gramalingam) [inline onnx/version_converter/adapters/lppool_17_18.h:19]: I guess this file is not needed? `convert.h` seems to use a CompatibleAdapter, which seems correct.

**Comment by gramalingam** [comment]
Agree with @kraiskil ... adding the test-cases would be useful.

**Comment by gramalingam** [comment]
Why not add the `dilations` attribute as well?

File: onnx/version_converter/adapters/lppool_17_18.h (hunk 1 above)
Comment (p-wysocki) [reply onnx/version_converter/adapters/lppool_17_18.h]: Good catch, I deleted the file and used `registerAdapter("LpPool", 17, 18, SetAttribute(kceil_mode, 0));` instead.

**Comment by p-wysocki** [comment]
@kraiskil @gramalingam I'm working on both adding tests and adding `dilations` attribute.

File: onnx/backend/base.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ class BackendRep:
     BackendRep to retrieve the corresponding results.
     """
 
-    def run(self, inputs: Any, **kwargs: Any) -> Tuple[Any, ...]:
+    def run(self, inputs: Any, **kwargs: Any) -> Tuple[Any, ...]:  # type: ignore
```
Comment (p-wysocki) [inline onnx/backend/base.py:63]: I had to ignore this line in `mypy` to pass linter check. It was throwing
`onnx/backend/base.py:63:5: error: Missing return statement  [empty-body]`
Is this a known issue? I did not make changes to this file. If you want me to fix it please let me know.

Comment (gramalingam) [reply onnx/backend/base.py:63]: I see Jacky has a PR: https://github.com/onnx/onnx/pull/4643

Comment (jcwchen) [reply onnx/backend/base.py:63]: Just merged. Updating the latest main branch in this PR should fix this issue. Thanks!

**Review by gramalingam — APPROVED** [review]

**Pull request merged**